app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -62,8 +52,19 @@ def valid_auth_tokens
     def analytics_properties
       {
         rendered_event_count: security_event_tokens.keys.count,
-        set_errors: security_event_token_errors,
+        timestamp: timestamp.iso8601,
+        success: timestamp.present?,
       }
     end
+
+    def timestamp
+      time = Time.zone.parse(params.require(:timestamp))
+
+      return nil unless time.time_zone.name == 'UTC'
+
+      time
```

Comment:
Is this a guard against which zone `Time.zone` is? I think we could be more explicit (also the zonebie randomized time zone in tests might confuse this)

```suggestion
      ActiveSupport::TimeZone['UTC'].parse(params.require(:timestamp))
```

Reply:
It's a guard against allowing specifying other time zones as the spec requires the timestamp be in UTC

Reply:
`ActiveSupport::TimeZone['UTC'].parse` would convert non-UTC timestamps into UTC, which it seems like we do not want to support based on the spec

Reply:
But `Time.zone` is shorthand for `ActiveSupport::TimeZone['UTC']` (or whatever zonebie sets it to in tests). 

I think if we want to guard against other timezones we might want to check the un-parsed string that it ends with for `/Z$/` or `+00(:00)?$/`

Reply:
Ah yep, thank you for explaining that, I misunderstood how it worked.  `ActiveSupport::TimeZone['UTC']` makes sense.  The formatting strictness is relatively minor, can revisit if needed I think?

Reply:
Yup! I agree. I think that the goal is to interpret stuff as UTC, and so it seems natural to convert inputs on the way in, as that code does